src/main/java/igym/services/GymService.java

```diff
@@ -24,12 +27,23 @@ public Gym createGym(Gym gym) {
         if (gymRepository.existsByName(gym.getName())) {
             throw new DuplicateGymException("A gym with the name '" + gym.getName() + "' already exists.");
         }
-        
+
         return gymRepository.save(gym);
     }
 
     public List<Gym> findAllGyms() {
         return gymRepository.findAll();
     }
 
+    public void deleteGym(UUID id) {
+        Gym gym = (gymRepository.findById(id))
+                .orElseThrow(() -> new GymNotFoundException("Gym with id " + id + " not found."));
+
+        if (gym.getStatus() == Status.inactive)
+            throw new GymNotFoundException("Gym with id " + id + " not found.");
```

Comment:
I would rewrite this method to make it more readable and compact

```suggestion
        Gym gym = (gymRepository.findById(id));
        if (gym == null || gym.getStatus() == Status.inactive)
            throw new GymNotFoundException("Gym with id " + id + " not found.");
```